### PR TITLE
Mention we need to update github action when we work on a new major release

### PR DESCRIPTION
**Pull request by tsteur**
Not sure when we would update that action but maybe when we start working on that new release.

There's a step `matomo-package build script` @sgiehl can this be removed now?
image

### Description:

Please include a description of this change and which issue it fixes. If no issue exists yet please include context and what problem it solves.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by sgiehl** [comment]
> There's a step matomo-package build script @sgiehl can this be removed now?

@tsteur yes. that one can be fully removed. I have refactored all the code so it is no longer required to set the major version somewhere. The current major version is always detected by looking at https://builds.matomo.org/LATEST

Btw. the release action doesn't need to be adjusted neither. The check for valid branches matches again `[4-9].x-dev`. So as long as we don't want to release Matomo 10.0, no change will be needed there.

**Comment by tsteur** [comment]
👍  removed that section, should be good to merge now or is there anything else we need to change @sgiehl ?

File: docs/release-management.md [hunk 1]
```diff
@@ -63,6 +63,7 @@ Also, as soon as we start working on the next major version, we need to require
 * Travis CI, see what we did for Matomo 3 in https://github.com/matomo-org/matomo/pull/8452#discussion-diff-35731015L93 and https://github.com/matomo-org/travis-scripts/pull/53
 * Create the `PULL_REQUEST_TEMPLATE` similar to https://github.com/matomo-org/matomo/pull/12412/files
 * Once the build for 4.x-dev succeeds, make it the default branch for Matomo and all plugins
+* Update the [submodule github action](https://developer.matomo.org/guides/release-management#when-starting-to-work-on-the-next-major-release) to use the new main branch
```
Comment (sgiehl) [inline docs/release-management.md]: Actually there are more than the `submodules update` action that need to be adjusted. But guess we can also have a look if it's easy to change the actions so they are using the configured default branch...

Comment (tsteur) [reply docs/release-management.md]: @sgiehl which other ones would we need to adjust? We could simply document them here quick and then it be easy to adjust them when needed

Comment (sgiehl) [reply docs/release-management.md]: [submodule action](https://github.com/matomo-org/matomo/blob/4.x-dev/.github/workflows/submodules.yml#L28)
[composer update action](https://github.com/matomo-org/matomo/blob/4.x-dev/.github/workflows/composer-update.yml#L28)
And the CLDR action as soon as it was merged.

**Pull request merged**